cargo/cargo_build_script.bzl

```diff
@@ -57,14 +54,7 @@ def _build_script_impl(ctx):
         stderr = ctx.actions.declare_file(ctx.label.name + ".stderr.log"),
     )
 
-    crate_name = ctx.attr.crate_name
-
-    # Derive crate name from the rule label which is <crate_name>_build_script if not provided.
-    if not crate_name:
-        crate_name = ctx.label.name
-        if crate_name.endswith("_build_script"):
-            crate_name = crate_name.replace("_build_script", "")
-        crate_name = crate_name.replace("_", "-")
+    pkg_name = _name_to_pkg_name(ctx.attr.name)
```

Comment:
I'm new to Bazel, but most places seem to use `ctx.label.name` rather than `ctx.attr.name`. I _think_ they should be equivalent, but is there any reason to prefer one over the other?

Reply:
These should be exactly equivalent :) I've switched to `ctx.label.name` for consistency